keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -431,11 +431,12 @@ public double product() {
 
     @Override
     public DoubleTensor slice(int dimension, int index) {
-        if (dimension == 0 && index == 0) {
-            return duplicate();
-        } else {
-            throw new IllegalStateException("Slice is only valid for dimension and index zero in a scalar");
+        if (dimension == 0 || dimension == 1) {
+            if (index == 0) {
+                return duplicate();
+            }
```

Comment:
I needed to allow slicing on dimension 1.

Note that our "scalars" are always rank 2 (i.e. their shape is 1x1) so this will not fail.